Fix: handle SHD calculation with isolated nodes

yashnator opened the pull request:
Ensures that we account for isolated nodes in the SHD calculation

### Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure you are making a pull request against the **dev branch** (left side). Also you should start *your branch* off *our dev*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.

### Issue number(s) that this pull request fixes
- Fixes #2094

### List of changes to the codebase in this pull request
- Modifies the SHD method `metrics/metrics.py` to handle isolated notes
- Adds a test to`tests/test_metrics/test_metrics.py`

Nimish-4 commented:
The formatting checks are failing. You can take a look into `.pre-commit-config.yaml` , and configure `black`, `isort` etc. as pre commit hooks to pass the formatting checks. Could also just run them manually from the terminal.

yashnator commented:
@Nimish-4 I fixed the linting errors. Let me know if there are more changes

Nimish-4 commented on pgmpy/metrics/metrics.py:
```diff
@@ -439,9 +439,11 @@ def SHD(true_model, est_model):
     nodes_list = true_model.nodes()
 
     dag_true = nx.DiGraph(true_model.edges())
+    dag_true.add_nodes_from(list(nx.isolates(true_model)))
```
Can you add a test case for this opposite scenario i.e. isolated nodes in the true graph compared to the estimated graph? And maybe another tougher test case that combines a few different scenarios? Looks fine beyond that.

yashnator committed:
Fix: handle SHD calculation with isolated nodes

Ensure isolated nodes are accounted for in the SHD
calculation

Fixes #2094

yashnator commented:
@Nimish-4 I've addressed the changes you requested. 
I've added a check to test SHD with graphs swapped for the opposite scenario, and added a slightly larger example with reversed edges, missing edges and isolated nodes. Could you take another look?

Nimish-4 commented:
@yashnator Looks good, thank you for the fix. @ankurankan will approve the changes.

ankurankan committed:
Simplifies tests

ankurankan commented:
Thanks for the PR @yashnator and the review @Nimish-4. Looks good to me. I have added a commit to simplify the tests.
